supplementary/alica_ros_turtlesim/src/base.cpp

```diff
@@ -18,14 +18,18 @@ namespace turtlesim
 {
 
 Base::Base(ros::NodeHandle& nh, ros::NodeHandle& priv_nh, const std::string& name, const std::string& roleset, const std::string& master_plan,
-        const std::string& path)
+        const std::string& path) : spinner(0)
 {
+    // Start spinner before initializing alica context
+    spinner.start();
```

Comment:
Spinner can still start after constructing AlicaContext  and before calling ac->init. A running spinner can start calling callbacks in the AlicaRosCommunication before other objects are constructed - set in the engine. Also other  applications will use this as a guideline so better do it at the right place.